### DISCUSSION TITLE
thick_mode=False actually starts thick mode as well

### DISCUSSION THREAD

**Discussion by BayerSe** [sqlalchemy/sqlalchemy#9291] [Usage Questions] [answered]
I'm not sure if this can be considered a bug or not, but setting `thick_mode=False` when using the Oracle driver starts it in thick mode nevertheless.

To check, run:

```python
with engine.connect() as conn:
    print(conn.execute(text("SELECT UNIQUE CLIENT_DRIVER FROM V$SESSION_CONNECT_INFO WHERE SID = SYS_CONTEXT('USERENV', 'SID')")).scalar())
```
which returns 
`python-oracledb thn : 1.2.2` when `engine = create_engine(url=url)` but `python-oracledb thk : 1.2.2` when the parameter `thick_mode` is set to any value, including `False`.

**Comment by CaselIT** [comment, marked as answer]
Hi,

It's working as expected, but I agree that the current behaviour is less than optimal. The dialect is currently checking for `is not None`. It probably should be changed with `not in (None, False)`

**Reply by CaselIT** [reply]
I've created  #9295 

for now you use `True` for on (or a dict with thick mode options) or `None` for off
